src/main/java/com/aws/greengrass/mqttclient/MqttClient.java

```diff
@@ -688,6 +688,7 @@ protected void runSpooler() {
                 break;
             } catch (Throwable ex) {
                 logger.atError().log("Unchecked error when publishing from spooler", ex);
+                throw ex;
```

Comment:
Correct. That was the behavior earlier also if you fall into the ExecutionException catch block.

Reply:
But that was only for when ExecutionException is due to InterruptedException which is expected. Now spooler can break even for intermittent errors and not recover until next Nucleus restart. Or am I missing something?

Reply:
Spooler would have earlier kept retrying if it received any runtime exception. This would have lead to infinite loop in case of a non-retryable error. We do not want that. If the exception happens because of some config issue, spooler start will be attempted again when the config update happens (does not require restart) since publish message will be triggered by ShadowListener and IotJobsHelper upon config update.